src/models/transferSettlement/facade.js

```diff
@@ -0,0 +1,305 @@
+        }
+        const participantPositionChangeRecords = participantPositionRecords.map(participantPositionRecord => {
+          participantPositionRecord.transferStateChangeId = transferStateChangeId[0].transferStateChangeId
+          return participantPositionRecord
+        })
+
+        await knex('participantPositionChange')
+          .insert(participantPositionChangeRecords)
+          .transacting(trx)
+
+        if (doCommit) {
+          await trx.commit
+        }
+      } catch (err) {
+        if (doCommit) {
+          await trx.rollback
+        }
+        throw err
+      }
+    }
+    if (trx) {
+      return await trxFunction(trx, false)
+    } else {
+      return await knex.transaction(trxFunction)
+    }
+  } catch (err) {
+    throw ErrorHandler.Factory.reformatFSPIOPError(err)
+  }
+}
+
+async function insertLedgerEntries (ledgerEntries, transferId, trx = null) {
+  Logger.info(`Ledger entries: ${JSON.stringify(ledgerEntries)}`)
+  try {
+    const knex = await Db.getKnex()
+    const trxFunction = async (trx, doCommit = true) => {
+      try {
+        for (const ledgerEntry of ledgerEntries) {
+          Logger.info(`Inserting ledger entry: ${JSON.stringify(ledgerEntry)}`)
+          await insertLedgerEntry(ledgerEntry, transferId, trx)
+        }
+        if (doCommit) {
+          await trx.commit
+        }
+      } catch (err) {
+        if (doCommit) {
+          await trx.rollback
+        }
+        throw err
+      }
+    }
+    if (trx) {
+      return await trxFunction(trx, false)
+    } else {
+      return await knex.transaction(trxFunction)
+    }
+  } catch (err) {
+    throw ErrorHandler.Factory.reformatFSPIOPError(err)
+  }
+}
+
+async function updateTransferSettlement (transferId, status, trx = null) {
+  Logger.info(Utility.breadcrumb(location, { method: 'updateTransferSettlement' }))
+  try {
+    const knex = await Db.getKnex()
+    const trxFunction = async (trx, doCommit = true) => {
+      try {
+        // Insert TransferParticipant ledger entry type.
+        await knex.from(knex.raw('transferParticipant (transferID, participantCurrencyId, transferParticipantRoleTypeId, ledgerEntryTypeId, amount)'))
+          .insert(function () {
+            this.from('transferParticipant AS TP')
+              .select('TP.transferId', 'TP.participantCurrencyId', 'TP.transferParticipantRoleTypeId', 'TP.ledgerEntryTypeId', knex.raw('?? * -1', ['TP.amount']))
+              .innerJoin('participantCurrency AS PC', 'TP.participantCurrencyId', 'PC.participantCurrencyId')
+              .innerJoin('settlementModel AS M', 'PC.ledgerAccountTypeId', 'M.ledgerAccountTypeId')
+              .innerJoin('settlementGranularity AS G', 'M.settlementGranularityId', 'G.settlementGranularityId')
+              .where(function () {
+                this.where({ 'TP.transferId': transferId })
+                this.andWhere(function () {
+                  this.andWhere({ 'G.name': 'GROSS' })
```

Comment:
'GROSS' should be set using an ENUM from central-services-shared.

Reply:
As discussed with @elnyry-sam-k: flag this issue for a future story If appropriate ENUMS are not already available for this.